common/attrs.c

```diff
@@ -143,18 +144,17 @@ attrs_build (CK_ATTRIBUTE *attrs,
 				free (add->pValue);
 			continue;
 
-		/* The attribute exitss, and we're overriding */
+		/* The attribute exists but we're overriding */
 		} else {
 			free (attr->pValue);
 		}
 
-		memcpy (attr, add, sizeof (CK_ATTRIBUTE));
-		if (!take_values && attr->pValue != NULL) {
-			if (attr->ulValueLen == 0)
-				attr->pValue = malloc (1);
-			else
-				attr->pValue = memdup (attr->pValue, attr->ulValueLen);
-			return_val_if_fail (attr->pValue != NULL, NULL);
+		if (take_values) {
```

Comment:
When is the ``take_values`` flag passed? Is it possible to have the flag set and the attribute to be added be an array?

Reply:
@ansasaki sorry for the long delay.

There are a couple of occasions where `take_values` is true: called from the internal API `p11_attrs_take` and `p11_attrs_merge`. Both of them take ownership of the operands, so even if they are an array, we don't need recursive copy; just moving the container should be sufficient.